viewing_party/party.py

```diff
@@ -0,0 +1,115 @@
+#Wave-1
+def create_movie(title, genre, rating):
+    if title and genre and rating:
+        movie_details = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie_details
+    else:
+        return None
+    
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):    
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)            
+    return user_data
+
+#Wave-2
+def get_watched_avg_rating(user_data):
+    user_watched_movie_rating = []
+    
+    for movie in user_data["watched"]:
+        user_watched_movie_rating.append(movie["rating"])
+        
+    if len(user_watched_movie_rating) == 0:
+        return 0.0
+    else:
+        average_rating = sum(user_watched_movie_rating) /len(user_watched_movie_rating)
+    return average_rating
```

Comment:
Even though this works well here is a suggestion to create this function without creating another list
```suggestion
def get_watched_avg_rating(user_data):
    if user_date["watched"] == 0
        return 0.0
    else:
        total_sum = 0
        for movie in user_data["watched"]
              total_sum += movie["rating"]
        average_rating = total_sum/len(user_data["watched"])
        return average_rating
```